main/src/modules/jointsAwareness/src/jointsAwarness.cpp

```diff
@@ -0,0 +1,413 @@
+    read_ObjLoc_Port.open(read_ObjLoc_PortName);
+
+    string objLocStreamPortName = "/iol2opc/objLoc:o";
+
+    unsigned int counter = 0;
+    bool isConnected = false;
+    while(!isConnected && counter < 3){
+        yInfo() << "Trying to connect to " << read_ObjLoc_PortName ;
+        isConnected = Network::connect(objLocStreamPortName, read_ObjLoc_PortName);
+        counter++;
+        Time::delay(1.0);
+    }
+    if(!isConnected){
+               yWarning() << "The port " << objLocStreamPortName << " from iol2opc is needed to stream the 2D proj of objects! jointsAwareness will NOT be able to stream this inforations" ;
+               isObjectStreamed = false;
+    }
+
+    if(isObjectStreamed == true){
+        string write_Obj2DProj_PortName = "/" + moduleName + "/" + "objects" + "/" + cameraSuffix + "/objects2DProj:o";
+        write_Obj2DProj_Port.open(write_Obj2DProj_PortName);
+    }
+
+    //rpc port
+    rpcPort.open(("/" + moduleName + "/rpc").c_str());
+    attach(rpcPort);
+
+    if(!bEveryThingisGood){
+        yError() << "Some dependencies/polydriver/options are not present/valid! Closing jointsAwareness!" ;
+        return false ;
+    }
+
+    yInfo() << "\n \n" << "----------------------------------------------" << "\n \n" << moduleName << " ready and streaming!!!!! \n \n ";
+
+    return true;
+}
+
+bool jointsAwareness::respond(const Bottle& command, Bottle& reply) {
+    string helpMessage = string(getName().c_str()) +
+        " commands are: \n" +
+        "help \n" +
+        "quit \n" ;
+
+    reply.clear();
+
+    if (command.get(0).asString() == "quit") {
+        reply.addString("quitting");
+
+        rpcPort.reply(reply);
+        return false;
+    } else {
+        cout << helpMessage;
+        reply.addString("ok");
+    }
+
+    rpcPort.reply(reply);
+
+    return true;
+}
+
+
+/* Called periodically every getPeriod() seconds */
+bool jointsAwareness::updateModule() {
+
+    isTorsoDone = false;
+
+    isTorsoDone = streamCartesian("left_arm");
+    isTorsoDone = streamCartesian("right_arm");
+
+    if(isObjectStreamed == true){
+        streamObjects();
+    }
+
+    return true;
+}
+
+void jointsAwareness::streamObjects() {
+
+    Bottle *b = read_ObjLoc_Port.read(false);
+    if (b!=NULL) {
+      // data received in *b
+
+        Bottle& bObjProj = write_Obj2DProj_Port.prepare();
+        bObjProj.clear();
+
+        //bottle from the port: (objName ("x y z")) (objName ("x y z"))
+        for(int i = 0; i < b->size(); i++){
+
+            Bottle bCurrentObject;
+            Vector vObjLoc;
+            Vector projection2D;
+            string objName = b->get(i).asList()->get(0).toString();
+            string s_objLoc = b->get(i).asList()->get(1).toString();
+            string s_objProj;
+
+            //yDebug() << "Object " << objName << " with loc: " << s_objLoc;
+
+            Bottle bCurrentObjectLoc;
+
+            std::string::size_type sz; // alias of size_t
```

Comment:
Looks like this is only used as optional argument, and then never used. Can be deleted.